### DISCUSSION TITLE
Pillow doesn't work in lambda

### DISCUSSION THREAD

**Discussion by jasonrdunne** [python-pillow/Pillow#7591] [Q&A] [answered]
Importing pillow from a lambda layer like so:
```python
from PIL import Image
```
Gives this error: `[ERROR] Runtime.ImportModuleError: Unable to import module 'lambda_function': cannot import name '_imaging' from 'PIL' (/opt/python/lib/python3.8/site-packages/PIL/__init__.py)`
Doesn't matter what version of python or pillow. I tested it with 3.8+7.2.0, 3.8+10.1.0 and 3.11+10.1.0, didn't make a difference. How is something as basic as importing Image being a nightmare?

**Comment by radarhere** [comment]
Is this helpful? It's some old AWS documentation on how to use Pillow in a lambda - https://web.archive.org/web/20210126181708/https://docs.aws.amazon.com/lambda/latest/dg/with-s3-example-deployment-pkg.html

![Screenshot 2023-12-01 at 4 36 53 pm](https://github.com/python-pillow/Pillow/assets/3112309/a2a5207e-dd58-4aad-9e53-df8bc47a7f03)

**Reply by radarhere** [reply, marked as answer]
It is not uncommon for users to come here with installation problems like this. The cause of these problems is that Pillow is not a pure Python library, but contains C code as well. The standard first thing to try is to uninstall Pillow and install it again. That may not apply in your case.

https://stackoverflow.com/questions/57197283/aws-lambda-cannot-import-name-imaging-from-pil also suggests
- [unzipping the relevant Pillow wheel and uploading those to your lambda](https://stackoverflow.com/questions/57197283/aws-lambda-cannot-import-name-imaging-from-pil/62121096#62121096), or
- using https://github.com/keithrozario/Klayers

**Reply by jasonrdunne** [reply]
Thanks, klayers does the trick, specifically this arn:  arn:aws:lambda:us-east-1:770693421928:layer:Klayers-p38-Pillow:10
